src/Core/Services/OpenAPI/OpenApiDocumentor.cs

```diff
@@ -431,6 +438,115 @@ private OpenApiOperation CreateBaseOperation(string description, List<OpenApiTag
             return operation;
         }
 
+        /// <summary>
+        /// This method creates a list of OpenApiParameter objects for the input parameters of a stored procedure.
+        /// A input parameter will be marked REQUIRED if default value is not available.
+        /// </summary>
+        private static List<OpenApiParameter> CreateStoredProcedureInputParameters(StoredProcedureDefinition spDefinition)
+        {
+            List<OpenApiParameter> parameters = new();
+
+            foreach ((string paramKey, ParameterDefinition parameterDefinition) in spDefinition.Parameters)
+            {
+                parameters.Add(
+                    GetOpenApiQueryParameter(
+                        name: paramKey,
+                        description: "Input parameter for stored procedure arguments",
+                        required: false,
+                        type: TypeHelper.GetJsonDataTypeFromSystemType(parameterDefinition.SystemType).ToString().ToLower()
+                    )
+                );
+            }
+
+            return parameters;
+        }
+
+        /// <summary>
+        /// Creates a list of OpenAPI parameters for querying tables and views.
+        /// The query parameters include $select, $filter, $orderby, $first, and $after, which allow the user to specify which fields to return,
+        /// filter the results based on a predicate expression, sort the results, and paginate the results.
+        /// </summary>
+        /// <returns>A list of OpenAPI parameters.</returns>
+        private static List<OpenApiParameter> CreateTableAndViewQueryParameters()
+        {
+            List<OpenApiParameter> parameters = new();
+
+            // Add $select query parameter
+            parameters.Add(
+                GetOpenApiQueryParameter(
+                    name: RequestParser.FIELDS_URL,
```

Comment:
Do we want to couple this class directly to the RequestParser class's implementation of how it stores the names of query parameters? This will now make the RequestParser class responsible for providing those names to other classes. Do we want to have it doing that, and if so, should we just be exposing them as public fields?

Reply:
i think request parser is the right place for the query names, because they're specifically used there, openAPI params needs the same queries, it's okay to source them from there.